File: DSA Practice/LeetCode/2090. K Radius Subarray Averages/solution.java
class Solution {
    public int[] getAverages(int[] nums, int k) {
        int n = nums.length;
        int[] res = new int[n];

        for (int i = 0; i < n; i++){
            res[i] = -1;
        }

        if (n < (2*k) + 1){
            return res;
        }

        int left = 0;
        int right = 0;
        long current_sum = 0;
        int d = 2*k + 1;

        while(right < n){
            current_sum += nums[right];
            if (right - left == 2*k){
                res[(int)((right+left)/2)] = (int)(current_sum / d);
                current_sum -= nums[left];
                left++;
            }

            right++;
        }

        return res;
    }
}

File: DSA Practice/LeetCode/2090. K Radius Subarray Averages/solution.kt
class Solution {
    fun getAverages(nums: IntArray, k: Int): IntArray {
        var n = nums.size
        var res : IntArray = IntArray(n) {-1}

        if (n < (2*k) + 1){
            return res
        }

        var left : Int = 0
        var right : Int = 0
        var current_sum : Long = 0L
        val d : Int = (2*k) + 1

        while (right < n){
            current_sum += nums.get(right)

            if (right - left == 2*k){
                res.set(((right + left) / 2).toInt(), (current_sum / d).toInt())
                current_sum -= nums.get(left)
                left++
            }

            right++
        }

        return res
    }
}